packages/web-components/src/dialog/dialog.ts

```diff
@@ -79,76 +32,15 @@ export class Dialog extends FASTElement {
    * @public
    * The type of the dialog modal
    */
-  @attr({ attribute: 'modal-type' })
-  public modalType: DialogModalType = DialogModalType.modal;
-
-  /**
-   * @public
-   * Indicates whether the dialog is open
-   */
-  @attr({ mode: 'boolean' })
-  public open: boolean = false;
-
-  /**
-   * @public
-   * Indicates whether the dialog has a title action
-   */
-  @attr({ mode: 'boolean', attribute: 'no-title-action' })
-  public noTitleAction: boolean = false;
-
-  /**
-   * @private
-   * Indicates whether focus should be trapped within the dialog
-   */
-  private trapFocus: boolean = false;
-
-  /**
-   * @public
-   * Method called when the 'open' attribute changes
-   */
-  public openChanged(oldValue: boolean, newValue: boolean): void {
-    if (newValue !== oldValue) {
-      if (newValue && !oldValue) {
-        this.show();
-      } else if (!newValue && oldValue) {
-        this.hide();
-      }
-    }
-  }
-
-  /**
-   * @public
-   * Method called when the 'modalType' attribute changes
-   */
-  public modalTypeChanged(oldValue: DialogModalType, newValue: DialogModalType): void {
-    if (newValue !== oldValue) {
-      if (newValue == DialogModalType.alert || newValue == DialogModalType.modal) {
-        this.trapFocus = true;
-      } else {
-        this.trapFocus = false;
-      }
-    }
-  }
-
-  /**
-   * @public
-   * Method to set the component's state based on its attributes
-   */
-  public setComponent(): void {
-    if (this.modalType == DialogModalType.modal || this.modalType == DialogModalType.alert) {
-      this.trapFocus = true;
-    } else {
-      this.trapFocus = false;
-    }
-  }
+  @attr({ attribute: 'type' })
+  public type: DialogType = DialogType.modal;
 
   /**
    * @public
    * Method to emit an event when the dialog's open state changes
-   * @param dismissed - Indicates whether the dialog was dismissed
    */
-  public onOpenChangeEvent = (dismissed: boolean = false): void => {
-    this.$emit('onOpenChange', { open: this.dialog.open, dismissed: dismissed });
+  public onOpenChangeEvent = (): void => {
+    this.$emit('onOpenChange', { open: this.dialog.open });
```

Comment:
Should we change the signature here and simplify to align with the dialog itself? `this.$emit('open', { open: this.dialog.open })`

We can probably update the name itself (nitpicky, I know)...perhaps `emitOpenChange`? Other ideas?

Reply:
So unexpectedly, `HTMLDialogElement` only emits a `close` and a `cancel` event but not an `open` event 😵‍💫 so we can't exactly align, but I think it's worth shortening because `onOpenChange` is a bit of a mouthful.

Reply:
Perfect? `open` wouldn't collide with anything there 😃